### PR TITLE
Langchain samples

### PR DESCRIPTION
**Pull request by paul-paliychuk**

### CODE REVIEW

File: zep_python/zep_client.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from zep_python.message.client import MessageClient
 from zep_python.user.client import UserClient
 
-API_URL = "https://api.getzep.com"
+API_URL = "https://api.development.getzep.com"
```
Comment (danielchalef) [inline zep_python/zep_client.py:17]: Rather set an environment variable than changing SDK code.

File: zep_python/memory/client.py [hunk 2]
```diff
@@ -829,16 +830,19 @@ def synthesize_question(self, session_id: str, last_n: int = 3) -> str:
...
-
+        print("Inside synthesize_question")
         params = {"lastNMessages": last_n}
         response = self.client.get(
             f"/sessions/{session_id}/synthesize_question", params=params
         )
 
         handle_response(response)
 
+        print("response.json()", response.json())
```
Comment (danielchalef) [inline zep_python/memory/client.py:844]: suggest using a debugger and not adding debug code to SDK. Less risk of leaving debug prints in merges.

File: zep_python/document/collections.py [hunk 3]
```diff
@@ -640,7 +640,7 @@ def search_return_query_vector(
 
     def search(
         self,
-        text: Optional[str] = None,
+        text: Optional[Union[str, Dict[str, str]]] = None,
```
Comment (danielchalef) [inline zep_python/document/collections.py:643]: why is this necessary?

File: zep_python/document/collections.py [hunk 4]
```diff
@@ -678,7 +678,7 @@ def search(
         """
 
         results, _ = self.search_return_query_vector(
-            text=text,
+            text=str(text),
```
Comment (danielchalef) [inline zep_python/document/collections.py:681]: why are you naively converting a possible dict to a string?

File: examples/langserve-sample/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+readme = "README.md"
+packages = [
+    { include = "app" },
+]
+
+[tool.poetry.dependencies]
+python = "^3.11"
+uvicorn = "^0.23.2"
+langserve = {extras = ["server"], version = ">=0.0.30"}
+pydantic = ">2"
```
Comment (danielchalef) [inline examples/langserve-sample/pyproject.toml:15]: perhaps limit to >=2 and <3?

File: examples/langserve-sample/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+message_history = {path = "packages/message_history", develop = true}
```
Comment (danielchalef) [inline examples/langserve-sample/pyproject.toml:16]: why are you installing these? and in develop mode?

File: examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    ConfigurableField,
+    RunnableParallel,
+)
+from langchain_core.runnables.utils import ConfigurableFieldSingleOption
+from langchain_openai import ChatOpenAI
+
+from zep_python import ZepClient
+from zep_python.langchain import ZepVectorStore
+
+ZEP_API_KEY = os.environ.get("ZEP_API_KEY", None)  # Required for Zep Cloud
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py:20]: I don't think `None` is necessary,  `get` by default returns `None` if the key is not found.

File: examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+ZEP_COLLECTION_NAME = os.environ.get("ZEP_COLLECTION", "langchaintest")
+
+if ZEP_API_KEY is None:
+    raise ValueError(
+        "ZEP_API_KEY is required for Zep Cloud. "
+        "Remove this check if using Zep Open Source."
+    )
+
+zep = ZepClient(
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py:30]: Please add api_url as an optional value pulled from the environment. You can even comment it out and add a comment that it is required for Zep Open Source. You can use this mechanism to target the dev cluster when you're testing.

File: examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    docs: List[Document],
+    document_prompt: PromptTemplate = DEFAULT_DOCUMENT_PROMPT,
+    document_separator: str = "\n\n",
+):
+    doc_strings = [format_document(doc, document_prompt) for doc in docs]
+    return document_separator.join(doc_strings)
+
+
+_inputs = RunnableParallel(
+    {"question": lambda x: x["question"], "context": retriever | _combine_documents},
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py:92]: Where is the standalone question being generated? There's no runnable with a call to `question_synthesis`?

File: examples/langserve-sample/packages/rag_vector_store/pyproject.toml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+fastapi = "^0.104.0"
+sse-starlette = "^1.6.5"
+
+[tool.langserve]
+export_module = "rag_vector_store"
+export_attr = "chain"
+
+[tool.templates-hub]
+use-case = "rag"
+author = "Elastic"
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/pyproject.toml:22]: Oh! We should change this!

File: examples/langserve-sample/packages/rag_vector_store/pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+integrations = ["OpenAI", "Zep"]
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/pyproject.toml:23]: We should remove OpenAI here too, I guess

File: examples/langserve-sample/packages/rag_vector_store/pyproject.toml [hunk 12]
```diff
@@ -0,0 +1,28 @@
+[tool.poetry]
+name = "rag_vector_store"
+version = "0.0.1"
+description = "A RAG application built with Zep. Zep provides a VectorStore implementation to the chain."
+authors = ["Daniel Chalef <daniel@getzep.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/pyproject.toml:8]: I think we need to add zep-python here - perhaps commented out ?

File: examples/langserve-sample/packages/rag_vector_store/pyproject.toml [hunk 13]
```diff
@@ -0,0 +1,28 @@
+[tool.poetry]
+name = "rag_vector_store"
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/pyproject.toml:2]: why is this pyproject file necessary?

File: examples/langserve-sample/packages/rag_vector_store/ingest.py [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+from zep_python import ZepClient
+from zep_python.langchain.vectorstore import ZepVectorStore
+
+load_dotenv()
+
+SOURCE = "https://www.theguardian.com/environment/2023/sep/28/plastic-eating-bacteria-enzyme-recycling-waste"  # noqa: E501
+
+ZEP_API_URL = (
+    os.environ.get("ZEP_API_URL")  # only required if you're using Zep Open Source
+    or "https://api.development.getzep.com"
```
Comment (danielchalef) [inline examples/langserve-sample/packages/rag_vector_store/ingest.py:17]: please don't hard code this. Set it in your environment.

File: examples/langserve-sample/packages/message_history/pyproject.toml [hunk 15]
```diff
@@ -0,0 +1,28 @@
+[tool.poetry]
```
Comment (danielchalef) [inline examples/langserve-sample/packages/message_history/pyproject.toml:1]: why is this pyproject file necessary?

File: examples/langserve-sample/packages/message_history/pyproject.toml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+fastapi = "^0.104.0"
+sse-starlette = "^1.6.5"
+
+[tool.langserve]
+export_module = "message_history"
+export_attr = "chain"
+
+[tool.templates-hub]
+use-case = "rag"
+author = "Elastic"
```
Comment (danielchalef) [inline examples/langserve-sample/packages/message_history/pyproject.toml:22]: please change this.

File: examples/langserve-sample/packages/message_history/message_history/chain.py [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+        "ZEP_API_KEY is required for Zep Cloud. "
+        "Remove this check if using Zep Open Source."
+    )
+
+zep = ZepClient(
+    api_key=ZEP_API_KEY,
+)
+
+# RAG answer synthesis prompt
+template = """You are a west coast rapper, I want you to rap the answer to the following question:
```
Comment (danielchalef) [inline examples/langserve-sample/packages/message_history/message_history/chain.py:29]: change to something we can publish

File: examples/langserve-sample/app/server.py [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+app = FastAPI()
+
+
+@app.get("/")
+async def redirect_root_to_docs():
+    return RedirectResponse("/docs")
+
+
+# Edit this to add the chain you want to add
```
Comment (danielchalef) [inline examples/langserve-sample/app/server.py:15]: to add the chain you want to use

File: examples/langserve-sample/README.md [hunk 19]
```diff
@@ -0,0 +1,79 @@
+# langserve-sample
```
Comment (danielchalef) [inline examples/langserve-sample/README.md:1]: is all of this still relevant to our example?

File: examples/langchain_simple_demo.py [hunk 20]
```diff
@@ -8,6 +8,7 @@
 
 from zep_python import ZepClient
 from zep_python.langchain.history import ZepChatMessageHistory
+from zep_python.memory.models import Session
```
Comment (danielchalef) [inline examples/langchain_simple_demo.py:11]: Should we delete this example?

File: examples/langchain/README.md [hunk 21]
```diff
@@ -0,0 +1,18 @@
+# rag-conversation-zep
```
Comment (danielchalef) [inline examples/langchain/README.md:1]: is this duplicative of elsewhere?

File: examples/langchain/ingest.py [hunk 22]
```diff
@@ -0,0 +1,46 @@
+# Ingest Documents into a Zep Collection
```
Comment (danielchalef) [inline examples/langchain/ingest.py:1]: do we need to remove this?

File: examples/langchain/rag_conversation_zep/chain.py [hunk 23]
```diff
@@ -0,0 +1,178 @@
+import os
```
Comment (danielchalef) [inline examples/langchain/rag_conversation_zep/chain.py:1]: do we need to remove this? Or did you get it working?

**Review by danielchalef — CHANGES_REQUESTED** [review]

File: examples/langchain-langserve/README.md [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+necessary for your application.
+
+In the below example, we inject the `OPENAI_API_KEY` environment
+variable with the value set in my local environment
+(`$OPENAI_API_KEY`)
+
+We also expose port 8080 with the `-p 8080:8080` option.
+
+```shell
+docker run -e OPENAI_API_KEY=$OPENAI_API_KEY -p 8080:8080 my-langserve-app
```
Comment (petergarbers) [inline examples/langchain-langserve/README.md:78]:
```
➜ docker build . -t my-langserve-app
[+] Building 0.5s (13/13) FINISHED                                                                                                                        docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                                                      0.0s
 => => transferring dockerfile: 400B                                                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                                                         0.0s
 => => transferring context: 2B                                                                                                                                           0.0s
 => [internal] load metadata for docker.io/library/python:3.11-slim                                                                                                       0.4s
 => [1/8] FROM docker.io/library/python:3.11-slim@sha256:53d6284a40eae6b625f22870f5faba6c54f2a28db9027408f4dee111f1e885a2                                                 0.0s
 => [internal] load build context                                                                                                                                         0.0s
 => => transferring context: 315B                                                                                                                                         0.0s
 => CACHED [2/8] RUN pip install poetry==1.6.1                                                                                                                            0.0s
 => CACHED [3/8] RUN poetry config virtualenvs.create false                                                                                                               0.0s
 => CACHED [4/8] WORKDIR /code                                                                                                                                            0.0s
 => CACHED [5/8] COPY ./pyproject.toml ./README.md ./poetry.lock* ./                                                                                                      0.0s
 => CACHED [6/8] RUN poetry install  --no-interaction --no-ansi --no-root                                                                                                 0.0s
 => CACHED [7/8] COPY ./app ./app                                                                                                                                         0.0s
 => CACHED [8/8] RUN poetry install --no-interaction --no-ansi                                                                                                            0.0s
 => exporting to image                                                                                                                                                    0.0s
 => => exporting layers                                                                                                                                                   0.0s
 => => writing image sha256:4a8edffb78821f20932d8b7c4191dd4bce25669fc8149cf9dde14810aaeeba04                                                                              0.0s
 => => naming to docker.io/library/my-langserve-app                                                                                                                       0.0s

What's Next?
  1. Sign in to your Docker account → docker login
  2. View a summary of image vulnerabilities and recommendations → docker scout quickview

zep-python/examples/langchain-langserve on  langchain-v2-example via 🐍 3.12.1 on 🐳 v24.0.6 via zep-python 
➜ docker images | grep my-lang        
my-langserve-app                                             latest         4a8edffb7882   About a minute ago   401MB

zep-python/examples/langchain-langserve on  langchain-v2-example via 🐍 3.12.1 on 🐳 v24.0.6 via zep-python 
➜ docker run -e OPENAI_API_KEY=$OPENAI_API_KEY -p 8080:8080 my-langserve-app
Traceback (most recent call last):
  File "/usr/local/bin/uvicorn", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/main.py", line 416, in main
    run(
  File "/usr/local/lib/python3.11/site-packages/uvicorn/main.py", line 587, in run
    server.run()
  File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 68, in serve
    config.load()
  File "/usr/local/lib/python3.11/site-packages/uvicorn/config.py", line 467, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/importer.py", line 24, in import_from_string
    raise exc from None
  File "/usr/local/lib/python3.11/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/code/app/server.py", line 4, in <module>
    from message_history_chain import chain as message_history_chain
ModuleNotFoundError: No module named 'message_history_chain'
```

File: examples/langchain-langserve/README.md [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+```shell
+export LANGCHAIN_TRACING_V2=true
+export LANGCHAIN_API_KEY=<your-api-key>
+export LANGCHAIN_PROJECT=<your-project>  # if not specified, defaults to "default"
+```
+
+## Launch LangServe
+
+```bash
+langchain serve
```
Comment (petergarbers) [inline examples/langchain-langserve/README.md:48]:
```
poetry install                                                  
Installing dependencies from lock file
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.

Package operations: 0 installs, 9 updates, 0 removals

  • Updating pydantic-core (2.14.6 -> 2.16.1)
  • Updating urllib3 (2.1.0 -> 2.2.0)
  • Updating pydantic (2.5.3 -> 2.6.0)
  • Updating langsmith (0.0.83 -> 0.0.85)
  • Updating aiohttp (3.9.1 -> 3.9.3)
  • Updating langchain-core (0.1.16 -> 0.1.18)
  • Updating langchain-community (0.0.16 -> 0.0.17)
  • Updating langchain (0.1.4 -> 0.1.5)
  • Updating langserve (0.0.40 -> 0.0.41)

Installing the current project: langserve-sample (0.1.0)

zep-python/examples/langchain-langserve on  langchain-v2-example via 🐍 3.12.1 on 🐳 v24.0.6 via zep-python took 5.9s 
➜ langchain serve
INFO:     Will watch for changes in these directories: ['/Users/pb/code/zepai/zep-python/examples/langchain-langserve']
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [3149] using StatReload
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/pb/code/zepai/zep-python/.venv/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
  File "/Users/pb/code/zepai/zep-python/.venv/lib/python3.12/site-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 684, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/pb/code/zepai/zep-python/.venv/lib/python3.12/site-packages/uvicorn/server.py", line 68, in serve
    config.load()
  File "/Users/pb/code/zepai/zep-python/.venv/lib/python3.12/site-packages/uvicorn/config.py", line 467, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pb/code/zepai/zep-python/.venv/lib/python3.12/site-packages/uvicorn/importer.py", line 24, in import_from_string
    raise exc from None
  File "/Users/pb/code/zepai/zep-python/.venv/lib/python3.12/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 994, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/pb/code/zepai/zep-python/examples/langchain-langserve/app/server.py", line 4, in <module>
    from message_history_chain import chain as message_history_chain
ModuleNotFoundError: No module named 'message_history_chain'

File: examples/langchain-langserve/app/message_history_chain.py [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+
+
+_inputs = RunnableParallel(
+    {
+        "question": lambda x: x["question"],
+        "chat_history": lambda x: x["chat_history"],
+    },
+).with_types(input_type=ChatHistory)
+
+chain = RunnableWithMessageHistory(
```
[the remark this answers is not in the document]
Comment (paul-paliychuk) [reply examples/langchain-langserve/app/message_history_chain.py:56]: We can inspect it in the playground or langsmith

File: examples/langchain-langserve/app/ingest.py [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+# Load
+loader = WebBaseLoader(SOURCE)
+data = loader.load()
+
+# Split
+text_splitter = RecursiveCharacterTextSplitter(chunk_size=400, chunk_overlap=200)
+all_splits = text_splitter.split_documents(data)
+
+# Add to vectorDB
+vectorstore = ZepVectorStore.from_documents(
```
[the remark this answers is not in the document]
Comment (paul-paliychuk) [reply examples/langchain-langserve/app/ingest.py:45]: Added some logs here

File: examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py (hunk 9 above)
Comment (paul-paliychuk) [reply examples/langserve-sample/packages/rag_vector_store/rag_vector_store/chain.py]: So we're not generating a question as this example doesn't have message history. We extract the question field as part of input for ChatPromptTemplate here.

File: zep_python/document/collections.py (hunk 3 above)
Comment (paul-paliychuk) [reply zep_python/document/collections.py]: Updated rag example to extract string from the dict before passing it to the retriever, my bad

File: zep_python/document/collections.py (hunk 4 above)
Comment (paul-paliychuk) [reply zep_python/document/collections.py]: Rolled back

File: examples/langchain-langserve/app/ingest.py [hunk 28]
```diff
@@ -0,0 +1,51 @@
...
+print("Split", len(all_splits), "documents")
+
+# Add to vectorDB
+vectorstore = ZepVectorStore.from_documents(
+    documents=all_splits,
+    collection_name=ZEP_COLLECTION_NAME,
+    zep_client=zep,
+)
+
+print("Added", len(all_splits), "documents to vectorDB")
```
Comment (danielchalef) [inline examples/langchain-langserve/app/ingest.py:51]: use f-string and fix spacing. Currently prints as "Added5documents"... Please use "Collection" when referring to Zep's vector indexes. We don't refer to Zep as a "vector database"

File: examples/langchain-langserve/app/message_history_chain.py [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+        "Remove this check if using Zep Open Source."
+    )
+
+zep = ZepClient(
+    api_key=ZEP_API_KEY,
+    api_url=ZEP_API_URL,  # only required if you're using Zep Open Source
+)
+
+# RAG answer synthesis prompt
+template = """I want you to answer to the following question, concisely to the best of your ability:"""
```
Comment (danielchalef) [inline examples/langchain-langserve/app/message_history_chain.py:32]: "Answer the user's question below. Be polite and helpful."

File: zep_python/memory/client.py [hunk 30]
```diff
@@ -831,7 +831,6 @@ def synthesize_question(self, session_id: str, last_n: int = 3) -> str:
         """
         if session_id is None or session_id.strip() == "":
             raise ValueError("session_id must be provided")
-
```
Comment (danielchalef) [inline zep_python/memory/client.py:834]: add back newline

File: zep_python/langchain/vectorstore.py [hunk 31]
```diff
@@ -351,7 +351,6 @@ def max_marginal_relevance_search(  # type: ignore # ignore inconsistent overrid
             raise ValueError(
                 "collection should be an instance of a Zep DocumentCollection"
             )
-
```
Comment (danielchalef) [inline zep_python/langchain/vectorstore.py:354]: add back newline

**Review by danielchalef — CHANGES_REQUESTED** [review]

**Review by danielchalef — APPROVED** [review]
lgtm
